osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/alarm/ClearAlarmRegisterCommandExecutorTest.java

```diff
@@ -0,0 +1,306 @@
+    when(this.dlmsConnection.set(this.setParameterArgumentCaptor.capture())).thenReturn(null);
+
+    this.assertExceptionAlarmRegister1(ProtocolAdapterException.class);
+  }
+
+  @Test
+  void connectionProblemAlarmRegister2() throws ProtocolAdapterException, IOException {
+    when(this.dlmsConnection.set(any(SetParameter.class)))
+        .thenReturn(AccessResultCode.SUCCESS)
+        .thenThrow(new IOException());
+
+    this.assertExceptionAlarmRegister2(ConnectionException.class);
+  }
+
+  @Test
+  void nullResultAlarmRegister2() throws ProtocolAdapterException, IOException {
+    when(this.dlmsConnection.set(this.setParameterArgumentCaptor.capture()))
+        .thenReturn(AccessResultCode.SUCCESS)
+        .thenReturn(null);
+
+    this.assertExceptionAlarmRegister2(ProtocolAdapterException.class);
+  }
+
+  @Test
+  void failureRegister1AndResultAlarmRegister2() throws ProtocolAdapterException, IOException {
+    when(this.dlmsConnection.set(this.setParameterArgumentCaptor.capture()))
+        .thenReturn(AccessResultCode.OTHER_REASON)
+        .thenReturn(AccessResultCode.SUCCESS);
+
+    final AccessResultCode accessResultCode = this.assertExceptionAlarmRegister1(null);
+    assertThat(accessResultCode).isEqualTo(AccessResultCode.OTHER_REASON);
+  }
+
+  @Test
+  void successRegister1AndResultAlarmRegister2() throws ProtocolAdapterException, IOException {
+    when(this.dlmsConnection.set(this.setParameterArgumentCaptor.capture()))
+        .thenReturn(AccessResultCode.SUCCESS)
+        .thenReturn(AccessResultCode.TEMPORARY_FAILURE);
+
+    final AccessResultCode accessResultCode = this.assertExceptionAlarmRegister2(null);
+    assertThat(accessResultCode).isEqualTo(AccessResultCode.TEMPORARY_FAILURE);
+  }
+
+  AccessResultCode assertExceptionAlarmRegister1(
+      final Class<? extends Exception> expectedExceptionClass)
+      throws ProtocolAdapterException, IOException {
+    final DlmsDevice dlmsDevice = new DlmsDevice("SMR 5.2 device");
+    dlmsDevice.setProtocol("SMR", "5.2");
+
+    when(this.dlmsObjectConfigService.getAttributeAddress(
+            dlmsDevice, DlmsObjectType.ALARM_REGISTER_1, null))
+        .thenReturn(
+            new AttributeAddress(
+                InterfaceClass.DATA.id(),
+                OBIS_CODE_ALARM_REGISTER_1,
+                DataAttribute.VALUE.attributeId()));
+
+    when(this.connectionManager.getDlmsMessageListener()).thenReturn(this.dlmsMessageListener);
+    when(this.connectionManager.getConnection()).thenReturn(this.dlmsConnection);
+
+    if (expectedExceptionClass != null) {
+      assertThrows(
+          expectedExceptionClass,
+          () -> {
+            this.executor.execute(
+                this.connectionManager, dlmsDevice, this.dto, this.messageMetadata);
+          });
+    } else {
+      return this.executor.execute(
+          this.connectionManager, dlmsDevice, this.dto, this.messageMetadata);
+    }
+    return null;
```

Comment:
Consider using `catchThrowable` to simplify this block of code.

Reply:
I changed it, but i don't see it is much simplified now.....